.clang-tidy

```diff
@@ -36,7 +36,7 @@ Checks: |
    readability-string-compare,
    readability-uniqueptr-delete-release,
    readability-function-size'
-WarningsAsErrors: '*'
+WarningsAsErrors: '*,-clang-analyzer-core.StackAddrEscapeBase,-clang-analyzer-optin.mpi.MPI-Checker'
```

Comment:
So the conclusion was that there is no smaller gun than excluding `core.StackAddrEscapeBase`? This is really unfortunate because this is almost always a bug, as it is in the place where the warning is caused in boost mpi...

Reply:
This should be fixed by the Boost version shipped in Ubuntu 20.04.

Reply:
Yes it's unfortunate, but Clang-Tidy offers no alternative for this warning. A conflicting priority here is that we need to be able to compile the code with the dev environment of the ICP machines, where we have the old Boost version (until we migrate to Ubuntu 20.08).

Reply:
Ok, @mkuron it's not sufficient that this is fixed on ubuntu 20.04, it needs to be fixed in all supported boost versions, which is not the case.

Reply:
> need to be able to compile the code with the dev environment of the ICP machines

Do you depend on warnings-as-errors there? As a stop-gap fix, you might just patch the header in /usr/include/boost locally for the time being. Frank already set up a test install of Ubuntu 20.04 today and it's looking pretty good, so upgrading the ICP is probably more a matter of weeks rather than months.

> it needs to be fixed in all supported boost versions, which is not the case

Does it? End users don't do clang-tidy. Since we require a specific version of clang-tidy anyway (whatever version is installed in our Docker container at any given time is likely the only version that gives zero warnings for the code base at the same time), it doesn't matter if we need a certain Boost version to get zero warnings.

Reply:
I agree this warning provides valuable information but we will have to wait for ICP computers to migrate to Ubuntu 20.04 before we can revert this change.

Reply:
If you stilll want to have a discussion about this, please open a ticket. I think we should go forward with this PR now.